### PR TITLE
Quote rpc route

### PR DESCRIPTION
**Pull request by bragov4ik**
#255
Return exchange route on `quote` rpc

### CODE REVIEW

File: pallets/liquidity-proxy/src/lib.rs [hunk 1]
```diff
@@ -709,7 +691,12 @@ impl<T: Config> Pallet<T> {
                     deduce_fee,
                 ),
             };
-            quote.map(|x| (x, atomic_path))
+            quote.map(|x| QuoteInfo {
```
Comment (pakrentos) [inline pallets/liquidity-proxy/src/lib.rs:694]: May be a good idea to add a short function for QuoteInfo which will build itself based upon QuoteOutcome

Comment (bragov4ik) [reply pallets/liquidity-proxy/src/lib.rs:694]: Not sure what is QuoteOutcome.

If you mean building from tuple as done in this line, then I don't think there is a need to make a separate function for this. such conversion (and any construction of struct `QuoteInfo`) occurs only here and `path` is added apart from the tuple

Comment (pakrentos) [reply pallets/liquidity-proxy/src/lib.rs:694]: @bragov4ik updated prev. comment; I mean add function like this
```
impl<T: Config> QuoteInfo<T> {
    pub fn from_parts(amount: QuoteAmount<T>, path: Vec<T::AssetId>) -> Self {
        Self {
            outcome: amount.0,
            rewards: amount.1,
            liquidity_sources: amount.2,
            path: path,
         }
    }
}
```
Just a suggestion related to code clarity.

Comment (bragov4ik) [reply pallets/liquidity-proxy/src/lib.rs:694]: Do you suggest to create a new struct for return value of `quote_pairs_with_flexible_amount`?

**Review by sevenzing — APPROVED** [review]

**Review by vovac12 — APPROVED** [review]

**Pull request merged**